### ISSUE TITLE
Swiping through widgets with Talkback changes the Tab on Android

### ISSUE THREAD

**Issue by bparrishMines** [flutter/flutter#19917]
When using a scrollable TabBar, Talkback will inconsistently swipe to an adjacent tab. I'm assuming we want similar behavior to iOS, where using the swipe gesture with Talkback only goes through widgets on the screen and won't cause swiping of tabs.

**Comment by jonahwilliams** [comment]
@bparrishMines which widget is this one?

**Comment by bparrishMines** [comment]
Its for the scrollable TabBar. I'll update the title and description.

**Comment by jonahwilliams** [comment]
I've found this can be fixed by *not* tagging the scrollable tab semantics node as a android.widget.ScrollView in the accessibility bridge. Given that there might also be some differences with android.widget.HorizontalScrollView - do we need to distinguish between scroll behaviors in our semantics tree? cc @goderbauer

**Comment by jonahwilliams** [comment]
Talkback does distinguish between a HorizontalScrollView and a TabBar (see https://github.com/google/talkback/blob/e69d4731fce02bb9e69613d0e48c29033cad4a98/compositor/src/main/java/Compositor.java#L575).  But if the difference is only in announcements, we might be able to get away with just using one of them and inferring as such from whether a node has scroll left/right actions

**Comment by jonahwilliams** [comment]
Possibly this too: https://github.com/google/talkback/blob/e69d4731fce02bb9e69613d0e48c29033cad4a98/utils/src/main/java/traversal/TraversalStrategyUtils.java#L417

**Comment by jonahwilliams** [comment]
In Summary:

We currently tag _all_ semantics node with a ScrollLeft, ScrollRight, ScrollUp, or ScrollDown action as an `android.widget.ScrollView`.  Talkback uses the className we provide to determine some basic a11y behaviors, such as whether to autoscroll the node.  By default, ScrollViews have auto scrolling behavior, while a TabBar widget wouldn't.

We need to distinguish between widgets that should have autoscrolling behavior and those that don't. perhaps via another (uggh) new semantics flag - useAutoScroll? or a scroll behavior enum... not sure.

**Comment by goderbauer** [comment]
Yeah we need to distinguish between scrollers that should scroll implicitly when the a11y focus move around and those that don't. I recently added `allowImplicitScrolling` to `ScrollPhysics` [1] which should encode which behavior we want for a given scroller. We probably have to read that property and somehow reflect it in the semantics tree via a flag.

[1] https://master-docs-flutter-io.firebaseapp.com/flutter/widgets/ScrollPhysics/allowImplicitScrolling.html
